python/ray/autoscaler/_private/util.py

```diff
@@ -222,6 +222,19 @@ def prepare_config(config: Dict[str, Any]) -> Dict[str, Any]:
     return with_defaults
 
 
+def translate_trivial_legacy_config(config: Dict[str, Any]):
+    """
+    Drop empty deprecated fields ("head_node" and "worker_node").
+    """
+
+    REMOVABLE_FIELDS = ["head_node", "worker_node"]
+
+    for field in REMOVABLE_FIELDS:
+        if field in config and not config[field]:
+            logger.warning(f"Detected and dropping the empty legacy field {field}.")
```

Comment:
Maybe something more prescriptive/informative like
`Dropping the empty legacy field {field}. {field} is not supported as of Ray 2.0.0. It is recommended to remove {field} from the cluster config.`